MFVNeutralino/test/predict2v_newsimpl.py

```diff
@@ -0,0 +1,175 @@
+print_pred_n2v_propagated_stat_err = True
+
+if data:
+    fn, presel_scale = 'JetHT%s.root' % year, 1.
+else:
+    fn, presel_scale = 'background_%s.root' % year, 1.
+
+def propagate_product(x, y, ex, ey):
+    p = x * y
+    e = p * ((ex / x)**2 + (ey / y)**2)**0.5
+    return e
+
+def fb(ft,efft,frt):
+    return (ft-frt)/(efft-frt)
+
+presel_f = ROOT.TFile(os.path.join(presel_path, fn))
+sel_f = ROOT.TFile(os.path.join(sel_path, fn))
+effs = pd.read_csv('MiniTree/efficiencies/all_effs.csv',index_col='variant')
+cb_vals = pd.read_csv('One2Two/cb_vals/cb_vals.csv',index_col='variant')
+
+fracdict = {}
+cdict = {}
+presel_var = 'presel_%s_%s' % (year, varname)
+fracdict['presel'] = fb(effs.at[presel_var,'ft'], effs.at[presel_var,'efft'], effs.at[presel_var,'frt'])
+
+for ntk in 3,4,5,7,8,9:
+    fracdict[ntk] = {}
+    cdict[ntk] = {}
+
+    var = '%strk_1v_%s_%s' % (ntk, year, varname)
+    try:
+        fracdict[ntk]['1v'] = fb(effs.at[var,'ft'], effs.at[var,'efft'], effs.at[var,'frt'])
+    except:
+        print "Warning: did not find effs for",ntk
+        fracdict[ntk]['1v'] = 0
+
+    year_formatted = year if not data else "data_%s" % year
+    cb_label = '%s_%strk' % (year_formatted, ntk)
+    try:
+        cdict[ntk]['cb'] = cb_vals.at[cb_label+'_cb','cb_val']
+        cdict[ntk]['cbbar'] = cb_vals.at[cb_label+'_cbbar','cb_val']
+    except:
+        print "Warning: did not find cb/cbbar for",ntk
+        cdict[ntk]['cb'] = 0
+        cdict[ntk]['cbbar'] = 0
+
+npresel, enpresel = get_integral(presel_f.Get('mfvEventHistosJetPreSel/h_npu'))
+npresel  *= presel_scale
+enpresel *= presel_scale
+f0 = fracdict['presel']
+
+print 'year:', year
+print 'presel events: %8.0f +- %4.0f' % (npresel, enpresel)
+print 'fraction of presel events with b-quarks: %.3f' % f0
+print '%16s %8s %8s %8s %19s %15s %35s' % ('n1v', 'f1', 'cb', 'cbbar', 'pred n2v', 'n2v', 'ratio')
+
+sum_n1v = 0
+alpha_nn = {}
+sum_rest = 0
+sum2_erest = 0
+sum_n2v = 0
+sum2_en2v =0
+sum2_en1v = 0
+for ntk in 3,4,5:
+   n1v, en1v = get_integral(sel_f.Get('%smfvEventHistosOnlyOneVtx/h_npu' % ('' if ntk == 5 else 'Ntk%s' % ntk)))
+   n2v, en2v = get_integral(sel_f.Get('%smfvEventHistosFullSel/h_npu' % ('' if ntk == 5 else 'Ntk%s' % ntk)))
+   sum_n1v += n1v
+   f1 = fracdict[ntk]['1v']
+   alpha = ((f0 - f1)*(f0 - f1)) - (f0*(f0-1)) #simplified alphas by assume Cb and Cbbar are about the same for all track multiplicities  
+   alpha_nn[ntk] = alpha
+   sum_rest += (n2v/alpha)
+   sum2_erest += (en2v/alpha)**2 
+   sum_n2v += n2v 
+   sum2_en2v += (en2v**2)
+   sum2_en1v += (en1v**2)
+print 'n1 = %8.0f'%(sum_n1v)
+print 'en1 = %f'%(math.sqrt(sum2_en1v)) 
+
+alpha_nm = {}
+for ntk in 'Ntk3or4','Ntk3or5', 'Ntk4or5':
+   tracks = [int(i) for i in ntk if i.isdigit()]
+   ntktot = sum(tracks)
+   f1 = []
+   for i, n in enumerate(tracks):
+        if n == 5:
+	    tracks[i] = ''
+	    f1.append(fracdict[5]['1v'])
+	else:
+	    tracks[i] = 'Ntk%s' % n
+	    f1.append(fracdict[n]['1v'])
```

Comment:
The indentations here look wrong. Not sure if it's a tab vs. space issue, but I would recommend turning on the settings that automatically turn tabs into spaces, so that it's consistent. It might actually work okay here, but definitely fix it + double check! (when I copied this into my text editor, it looked like the `else` was aligned with the `for`, which would definitely be bad.

Reply:
Okay, will fix this and check the results again.

Reply:
it does not seem to be an issue on my local editor and the one on GitHub though. `if` and `else` align as expected.

Reply:
They're not aligned--the if is indented more than the else! If it's tabs vs. spaces, it would be good to replace the tabs with spaces (especially since python depends so heavily on spacing).